pkg/volume/csi/csi_attacher.go

```diff
@@ -286,15 +288,25 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 	}
 
 	// Setup
-	if spec == nil {
-		return fmt.Errorf("attacher.MountDevice failed, spec is nil")
-	}
-	csiSource, err := getPVSourceFromSpec(spec)
+	volSrc, pvSrc, err := getSourceFromSpec(spec)
 	if err != nil {
-		klog.Error(log("attacher.MountDevice failed to get CSIPersistentVolumeSource: %v", err))
+		klog.Error(log("attacher.MountDevice failed to get volume source: %v", err))
 		return err
 	}
 
+	// CSIVolumeSource for ephemeral volumes not supported
+	// simply return quietly
+	// TODO (#75352) ability to skip DeviceMount when ephemeral volume
+	if volSrc != nil && utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
+		klog.Info(log("attacher.MountDevice skipping device mount from ephemeral CSIVolumeSource"))
+		return nil
```

Comment:
This should be OK.  If ephemeral is skipped during MountDevice, there should never be an orphaned path (for ephemeral) that needs reconstruction from it.  Otherwise, not skipping it or attempting to save metadata during MountDevice causes other problems. Mainly, there is no volumeHandle available for the ephemeral volume source at that time. Therefore metadata would be incomplete.

Reply:
Reconstruction applies to per-pod paths.  CSI stores metadata for volumes even if they don't support NodeStage.  But we don't have a pod name at this point to be able to uniquely identify an ephemeral volume?

For alpha I think it is fine to not support reconstruction, but we find a solution before going to beta.

Reply:
Ok